### PR TITLE
504-refactor: Fsd widget breadcrumbs

### LINKED ISSUE #504: FSD: Widget Breadcrumbs
## 📝 Describe the refactoring task

Part of the issue #347 
Breadcrumbs component refactor

## 📍 Code location

`src/widgets/breadcrumbs`

## ✅ Definition of Done (DoD):

1. Refactor `breadcrumbs.tsx`'s `scss` to `scss modules` including:

-   remove all parent selector `& .card-content` -> `.card-content`. 
-   change all possible values to variables if such exists.
-   rename all classes to be generic e.g `.nodejs-trainer 🙅‍♂️` -> `.trainer 👌`.
-   and everything in your opinion that needs to be refactored

2. refactor JSX to utilise `HTML5` tags
3. move unit test inside the `ui` folder
4. rename `breadcrumbs.types.ts` to `types.ts`
5. in the `breadcrumbs.tsx` file, inside `transformedCrumbs.map` replace ternary operator with the `if else` statement (or you can come up with simpler logic If you want)
6. remove `breadcrumbNameMap` re-export inside the `index.ts` file

### PR DESCRIPTION
**Pull request by thirdmadman**
## What type of PR is this? (select all that apply)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [x] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description

- [x] 1. Refactor `breadcrumbs.tsx`'s `scss` to `scss modules` including:
- [x] 1.1 remove all parent selector `& .card-content` -> `.card-content`. 
- [x] 1.2  change all possible values to variables if such exists.
- [x] 1.3 rename all classes to be generic e.g `.nodejs-trainer 🙅‍♂️` -> `.trainer 👌`.
- [x] 1.4 and everything in your opinion that needs to be refactored
- [x] 2. refactor JSX to utilise `HTML5` tags
- [x] 3. move unit test inside the `ui` folder
- [x] 4. rename `breadcrumbs.types.ts` to `types.ts`
- [x] 5. in the `breadcrumbs.tsx` file, inside `transformedCrumbs.map` replace ternary operator with the `if else` statement (or you can come up with simpler logic If you want)
- [x] 6. remove `breadcrumbNameMap` re-export inside the `index.ts` file

## Related Tickets & Documents

- Related Issue #504
- Closes #504

## Screenshots, Recordings

none

## Added/updated tests?

- [x] 👌 Yes
- [ ] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `Breadcrumb` component for improved navigation.
	- Added a new `BreadcrumbItem` component to enhance modularity in breadcrumb rendering.
	- Implemented styles for breadcrumb navigation to enhance visual presentation.
- **Bug Fixes**
	- Updated tests for the `Breadcrumbs` component to improve performance and clarity.
- **Refactor**
	- Restructured breadcrumb rendering logic for better modularity and maintainability.
	- Removed unnecessary exports and consolidated type definitions.
- **Tests**
	- Added unit tests for the `Breadcrumb` and `BreadcrumbItem` components to ensure correct functionality.

### CODE REVIEW

File: src/widgets/breadcrumbs/ui/breadcrumb.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const cx = classNames.bind(styles);
+
+interface BreadcrumbProps {
+  linkTo: string;
+  text: string;
+  isLastLink?: boolean;
+}
+
+export function Breadcrumb(props: BreadcrumbProps) {
+  const { linkTo, text, isLastLink } = props;
```
Comment (MielomankA) [inline src/widgets/breadcrumbs/ui/breadcrumb.tsx:15]:
```suggestion
export function Breadcrumb({ linkTo, text, isLastLink }: BreadcrumbProps) {
```
Maybe we can do destructuring here? similar to other components

Comment (thirdmadman) [reply src/widgets/breadcrumbs/ui/breadcrumb.tsx]: fixed in https://github.com/rolling-scopes/site/pull/516/commits/e49804e51e5eb37e7d12fe448dbabe83738e5535

**Comment by thirdmadman** [comment]
Im curious if it's better to rename ```Breadcrumbs``` -> ```BreadcrumbsNavigation```, so it's not overlaping with ```Breadcrumb```

**Review by Quiddlee — APPROVED** [review]

**Comment by Quiddlee** [comment]
> Im curious if it's better to rename `Breadcrumbs` -> `BreadcrumbsNavigation`, so it's not overlaping with `Breadcrumb`

Definitely need a better naming

**Review by MielomankA — APPROVED** [review]

**Review by SpaNb4 — APPROVED** [review]

**Pull request merged**